packages/devtools_app/lib/src/screens/memory/panes/chart/controller/chart_connection.dart

```diff
@@ -36,34 +36,86 @@ class ChartConnection extends DisposableController
   );
 
   Timer? _pollingTimer;
-  bool _connected = false;
 
-  late final isDeviceAndroid =
-      serviceConnection.serviceManager.vm?.operatingSystem == 'android';
+  /// If completed, this instance was connected to the application.
+  final Completer<void> _initialized = Completer();
+
+  void _stopConnection() {
+    _pollingTimer?.cancel();
+    cancelStreamSubscriptions();
+    cancelListeners();
+  }
+
+  // True if connection was started and then stopped.
+  bool get _isConnectionStopped {
+    assert(_initialized.isCompleted);
+    return _pollingTimer?.isActive == false;
+  }
+
+  late bool isDeviceAndroid;
+
+  /// True if DevTools is in connected mode.
+  ///
+  /// If DevTools is in offline mode, stops connection and returns false.
+  bool _checkConnection() {
+    assert(_initialized.isCompleted);
+    if (_isConnectionStopped) return false;
+
+    // If connection is up and running, return true.
+    if (!offlineDataController.showingOfflineData.value &&
```

Comment:
I want to stop listeners right away because I do not want to take events. Disconnect consists of many parts and it is not responsibility of this class to know or guarantee that events will not happen in case of disconnect. So, it just should stop listening.

Reply:
After discussion with Kenzie I realized my understanding of cycle is wrong. With updated understanding, yes, `dispose` is enough.